From d9cc4774e3a065e7ad7dec942cb3bdb6ac9032c2 Mon Sep 17 00:00:00 2001
From: Anna Henningsen <anna@addaleax.net>
Date: Wed, 11 Jul 2018 01:08:55 +0200
Subject: [PATCH] [platform] Work around potentially missing `madvise()`

Some Linux kernels are built without `madvise()`. In that
case, do not crash but rather ignore the return value instead.

Refs: https://github.com/nodejs/node/issues/21562
Change-Id: Iff8d0d0e9e7935804e2fc994be7f8bd21c553846
---

--- a/deps/v8/src/base/platform/platform-posix.cc
+++ b/deps/v8/src/base/platform/platform-posix.cc
@@ -156,6 +156,8 @@ int ReclaimInaccessibleMemory(void* addr
 #else
   int ret = madvise(address, size, MADV_FREE);
 #endif
+  if (ret != 0 && errno == ENOSYS)
+    return 0;  // madvise is not available on all systems.
   if (ret != 0 && errno == EINVAL) {
     // MADV_FREE only works on Linux 4.5+ . If request failed, retry with older
     // MADV_DONTNEED . Note that MADV_FREE being defined at compile time doesn't
